sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildInstaller.java

```diff
@@ -70,35 +76,56 @@ private File install(Location location, File toDir, boolean useOldScript) {
       }
       doInstall(zipFile, toDir, null);
     }
-    return locateInstalledScript(null, toDir, useOldScript);
+    return locateInstalledScript(null, toDir, useOldScript, useDotNetCore);
   }
 
-  private File install(Version scannerVersion, File toDir, boolean useOldScript) {
+  private File install(Version scannerVersion, File toDir, boolean useOldScript, boolean useDotNetCore) {
     clearCachedSnapshot(scannerVersion, toDir);
     if (!isInstalled(scannerVersion, toDir)) {
       LOG.info("Installing Scanner for MSBuild {}", scannerVersion);
-      File zipFile = locateZip(scannerVersion);
+      File zipFile = locateZip(scannerVersion, useDotNetCore);
 
       if (zipFile == null || !zipFile.exists()) {
         throw new IllegalArgumentException("Unsupported scanner for MSBuild version: " + scannerVersion);
       }
 
       doInstall(zipFile, toDir, scannerVersion);
     }
-    return locateInstalledScript(scannerVersion, toDir, useOldScript);
+    return locateInstalledScript(scannerVersion, toDir, useOldScript, useDotNetCore);
   }
 
   void doInstall(File zipFile, File toDir, @Nullable Version scannerVersion) {
     try {
       File scannerDir = new File(toDir, directoryName(scannerVersion));
       scannerDir.mkdirs();
       ZipUtils.unzip(zipFile, scannerDir);
+      if (SystemUtils.IS_OS_UNIX &&
+        scannerVersion != null
+        && scannerVersion.isGreaterThan(ScannerForMSBuild.DOT_NET_CORE_INTRODUCTION_VERSION)) {
+        // change permissions on binary files from sonar-scanner included in scanner
+        setSonarScannerBinariesAsExecutable(scannerDir);
+      }
     } catch (Exception e) {
       throw new IllegalStateException("Fail to unzip Scanner for MSBuild from " + zipFile + " to " + toDir, e);
     }
   }
 
-  private File locateZip(Version scannerVersion) {
+  private void setSonarScannerBinariesAsExecutable(File scannerDir) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "setSonarScannerBinariesAsExecutable" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.orchestrator%3Aorchestrator-parent&pullRequest=45&issues=AWMruZBBau6S9RpGg6aA&open=AWMruZBBau6S9RpGg6aA)

[ISSUE_KEY]: AWMruZBBau6S9RpGg6aA